### PR TITLE
add login with pin test

### PR DESCRIPTION
**Pull request by shravanmechineni**
https://tools.hmcts.net/jira/browse/SIDM-4093

Add login with pin test.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/js/uplift_user_test.js [hunk 1]
```diff
@@ -48,6 +50,27 @@ AfterSuite(async (I) => {
...
+    I.see('code=');
+
+    let pageSource = await I.grabSource();
+    let code = pageSource.match(/\?code=([^&]*)(.*)/)[1];
+    let accessToken = await I.getAccessToken(code, serviceName, TestData.SERVICE_REDIRECT_URI, TestData.SERVICE_CLIENT_SECRET);
+
+    let userInfo = await I.retry({retries: 3, minTimeout: 10000}).getOidcUserInfo(accessToken);
+    expect(userInfo.roles).to.eql(['letter-holder']);
+
+    I.resetRequestInterception();
```
Comment (tbamido) [inline src/test/js/uplift_user_test.js:71]: this line wouldn't run if one of the requests failed, move it to a catch block or an afterAll if there's no issue calling it multiple times

Comment (shravanmechineni) [reply src/test/js/uplift_user_test.js:71]: yeah, I'm not sure how we can do that.

Comment (shravanmechineni) [reply src/test/js/uplift_user_test.js]: Moved to teard own method.

File: src/test/js/shared/idam_helper.js [hunk 2]
```diff
@@ -505,7 +505,7 @@ class IdamHelper extends Helper {
         const helper = this.helpers['Puppeteer'];
         helper.page.setRequestInterception(true);
         helper.page.on('request', request => {
-            if (request.url().indexOf('/login') > 0 || request.url().indexOf('/register') > 0 || request.url().indexOf('/activate') > 0 || request.url().indexOf('/verification') > 0) {
+            if (request.url().indexOf('/login') > 0 || request.url().indexOf('/register') > 0 || request.url().indexOf('/activate') > 0 || request.url().indexOf('/verification') > 0 | request.url().indexOf('/pin') > 0) {
```
Comment (tbamido) [inline src/test/js/shared/idam_helper.js:508]: minor:
could use StringUtils.containsAny in here
see http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#containsAny-java.lang.CharSequence-java.lang.CharSequence...-

Comment (shravanmechineni) [reply src/test/js/shared/idam_helper.js:508]: Ok but that doc is in Java.

Comment (radoslaw-orlowski-amido) [reply src/test/js/shared/idam_helper.js:508]: @shravanmechineni : https://stackoverflow.com/questions/5582574/how-to-check-if-a-string-contains-text-from-an-array-of-substrings-in-javascript

Comment (shravanmechineni) [reply src/test/js/shared/idam_helper.js]: Done.

File: src/test/js/shared/idam_helper.js [hunk 3]
```diff
@@ -504,8 +504,10 @@ class IdamHelper extends Helper {
     interceptRequestsAfterSignin() {
         const helper = this.helpers['Puppeteer'];
         helper.page.setRequestInterception(true);
+        const pages = ["/login", "/register", "/activate", "/verification"];
+
         helper.page.on('request', request => {
-            if (request.url().indexOf('/login') > 0 || request.url().indexOf('/register') > 0 || request.url().indexOf('/activate') > 0 || request.url().indexOf('/verification') > 0 | request.url().indexOf('/pin') > 0) {
+            if (pages.some(v => request.url().includes(v))) {
```
Comment (tbamido) [inline src/test/js/shared/idam_helper.js:510]: lol I didn't realise it's javascript not java and recommended using stringutils 🤦‍♂️

**Pull request merged**